packages/compass-query-bar/src/stores/ai-query-reducer.ts

```diff
@@ -22,6 +22,7 @@ export type AIQueryState = {
   aiPromptText: string;
   status: AIQueryStatus;
   aiQueryFetchId: number; // Maps to the AbortController of the current fetch (or -1).
+  didSubmitFeedback: boolean;
```

Comment:
Don't have the strongest preference, but as this basically relies on query succeeding (state we already have in store!) and telemetry (`track` calls) for actual feedback submission, this whole thing including its state can live outside of the redux store? As I see it, you can track submission state inside the `QueryFeedback` component and (un)mounting it when ai query state changes should be enough to correctly reset it, so something like this:

```tsx
function AITextInput() {
  // ...

  {didSucceed && <QueryFeedback />}
}

...

function QueryFeedback() {
  const [didSubmit, setDidSubmit] = useState(false);
  
  const onSubmit = () => {
    track(...)
    setDidSubmit(true);
  }

  if (didSubmit) {
    return null; // ... or a fancy animation if we want to now that we control the rendering from here :)
  }
  
  // ...
}
```

I think this would have two big benefits: less state in redux store (always better if can be done!) and probably would be easier to share it with agg builder in the future

Reply:
Done! Also added an animation for when the feedback is submitted:

https://github.com/mongodb-js/compass/assets/1791149/cb659813-2e2b-4892-9f45-fcda8a4ca601